component/sections/pricing/variant_a.js

```diff
@@ -7,71 +7,79 @@ function VariantA({
   title,
   description,
   plans,
-  sanityToken,
   hashKey,
   apiVersion,
   stripeSecretKey,
   stripePKey,
   NEXT_PUBLIC_DXP_STUDIO_ADDRESS,
 }) {
   const [plan, setPlan] = React.useState("monthly");
-  const [subscriptionProducts, setSubscriptionProducts] = React.useState([]);
   const [usePlan, setUsePlan] = React.useState(plans);
-
-  async function getPriceId(plans) {
-    let plansResponse = [];
-    let i = 0;
-    for (; i < plans?.length; ) {
-      const payload = {
-        id: `dxpstudio-pricing-${plans[i]?._key}-${plans[i]?.planType?.replace(
-          / /g,
-          "-"
-        )}-recurring-monthlyPrice-${plans[i]?.monthlyPrice}-yearlyPrice-${
-          plans[i]?.yearlyPrice
-        }`,
-        sanityToken,
-        hashKey,
-        stripeSecretKey,
-        apiVersion,
-      };
-      try {
-        const response = await axios.post(
-          `${NEXT_PUBLIC_DXP_STUDIO_ADDRESS}/api/stripe-account/get-product-by-id`,
-          payload
-        );
-        const data = await response.data;
-        plansResponse.push(data.data);
-      } catch (error) {
-        console.log(error);
-      }
-      i++;
-    }
-    setSubscriptionProducts(plansResponse);
-    return plansResponse;
-  }
+  const [pKeyError, setPKError] = React.useState(false);
+  const comma = Intl.NumberFormat("en-us");
 
   React.useEffect(() => {
-    getPriceId(usePlan);
-  }, [plans]);
+    async function getPriceId(plans) {
+      let i = 0;
+      for (; i < plans?.length; ) {
+        const productPayload = {
+          credentials: {
+            hashKey,
+            stripeSecretKey,
+            apiVersion,
+          },
+          StripeParams: {
+            id: `dxpstudio-pricing-${plans[i]?._key}-${plans[
+              i
+            ]?.planType?.replace(/ /g, "-")}-recurring-monthlyPrice-${
+              plans[i]?.monthlyPrice
+            }-yearlyPrice-${plans[i]?.yearlyPrice}`,
+          },
+        };
```

Comment:
This payload needs to be updated as per https://github.com/webriq-pagebuilder/app/pull/63#discussion_r719403192